controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -236,22 +198,12 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 					cephClient := &rookCephv1.CephClient{}
 					cephClient.Name = cephResource.Name
 					cephClient.Namespace = r.namespace
-					if err := r.delete(cephClient); err != nil {
-						return ctrl.Result{}, fmt.Errorf("unable to delete CephClient : %v", err)
-					}
-				case "CephBlockPool":
-					cephBlockPool := &rookCephv1.CephBlockPool{}
-					cephBlockPool.Name = cephResource.Name
-					cephBlockPool.Namespace = r.namespace
-					if err := r.delete(cephBlockPool); err != nil {
-						return ctrl.Result{}, fmt.Errorf("unable to delete CephBlockPool : %v", err)
-					}
-				case "CephFilesystemSubVolumeGroup":
-					cephFilesystemSubVolumeGroup := &rookCephv1.CephFilesystemSubVolumeGroup{}
-					cephFilesystemSubVolumeGroup.Name = cephResource.Name
-					cephFilesystemSubVolumeGroup.Namespace = r.namespace
-					if err := r.delete(cephFilesystemSubVolumeGroup); err != nil {
-						return ctrl.Result{}, fmt.Errorf("unable to delete CephFilesystemSubVolumeGroup : %v", err)
+					_ = r.get(cephClient)
+					cephUserType := cephClient.Annotations[StorageCephUserTypeAnnotation]
+					if cephUserType == "healthchecker" {
+						if err := r.delete(cephClient); err != nil {
+							return ctrl.Result{}, fmt.Errorf("unable to delete CephClient : %v", err)
+						}
```

Comment:
As I already asked in the prev review, this needs to be reverted:
https://github.com/red-hat-storage/ocs-operator/pull/1603#discussion_r858097686

```suggestion
					if err := r.delete(cephClient); err != nil {
						return ctrl.Result{}, fmt.Errorf("unable to delete CephClient : %v", err)
					}
```

Reply:
one of these occurrences was removed,
 this was left.